hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/operations/JoinCheckOperation.java

```diff
@@ -61,12 +66,13 @@ public void run() {
             ILogger logger = getLogger();
             try {
                 if (service.getClusterJoinManager().validateJoinMessage(request)) {
-                    try {
-                        // also validate other node's version is able to join our cluster
-                        node.getNodeExtension().validateJoinRequest(request);
+                    // also validate other cluster's major.minor version is same as this cluster
```

Comment:
Not really, `NodeExtension.validateJoinRequest(request)` validates `JoinMessage.version` (ie. the joining node's codebase version) is OK to join the cluster. In the specific case, it would validate that the subcluster's master node codebase version is OK to join the other subcluster, but this does not provide any indication about our ability to merge one cluster to another. That is why it is replaced with validation of cluster-to-cluster version `major.minor` equality.